File: solutions/3510-maximize-the-total-height-of-unique-towers/maximize-the-total-height-of-unique-towers.java
class Solution {
    public long maximumTotalSum(int[] nums) {
        Arrays.sort(nums);
        long ans=nums[nums.length-1];
        for(int i=nums.length-2;i>=0;i--){
            if(nums[i]>=nums[i+1]){
                nums[i]=nums[i+1]-1;
                ans+=nums[i];
                if(nums[i]<=0){
                    return -1;
                }
            }
            else{
                ans+=nums[i];
            }
        }
        return ans;
    }
}

File: solutions/3510-maximize-the-total-height-of-unique-towers/maximize-the-total-height-of-unique-towers.cpp


class Solution {
public:
    long long maximumTotalSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        long long ans = nums.back();
        
        for (int i = nums.size() - 2; i >= 0; --i) {
            if (nums[i] >= nums[i + 1]) {
                nums[i] = nums[i + 1] - 1;
                ans += nums[i];
                if (nums[i] <= 0) {
                    return -1;
                }
            } else {
                ans += nums[i];
            }
        }
        
        return ans;
    }
};